### ISSUE TITLE
DatePicker: cannot type in datepicker when first displayed (14.0.0)

### ISSUE THREAD

**Issue by cocorossello** [primefaces/primefaces#11928]
### Describe the bug

Since PF 14.0.0, datepicker input seems is now writable for some reason (however it isn't readonly).  It is writeable after a date select event. 

### Reproducer

Reproducer in https://github.com/cocorossello/primefaces-test/

Code:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://xmlns.jcp.org/jsf/html"
      xmlns:p="http://primefaces.org/ui"
      xmlns:f="http://xmlns.jcp.org/jsf/core">
<f:view>
    <h:head></h:head>
    <h:body>
        <p:datePicker/>
    </h:body>
</f:view>
</html>

```

Just click in the input and type some keys. It used to work in 13.0.x

Mojarra/Myfaces does not make a difference

### Expected behavior

Input should be writeable

### PrimeFaces edition

Community

### PrimeFaces version

14.0.0

### JSF implementation

All

### JSF version

2.3-NEXT

### Java version

17

**Comment by melloware** [comment]
Is the popup open?  If so it's because of accessibility requirements that if the popup is open you are interacting with the popup and it has focus. When you close it then the input has focus and can be typed. 

This was all part of making DatePicker 508 and WCAG compliant.  

Can you verify that is your issue?

**Comment by melloware** [comment]
For reference: https://github.com/primefaces/primefaces/pull/10513

**Comment by melloware** [comment]
Just turn showOnFocus="false". That way it's typing when you come into the field and the popup you can open with keyboard or button.

**Comment by tandraschko** [comment]
is this really a WCAG requirement? how do it other libs?
very weird to disallow typing

**Comment by cocorossello** [comment]
But I still would like to see the calendar, it's useful.

See this one, the input still gets the focus on open but you can arrow down to go to the panel
https://hanshillen.github.io/jqtest/#goto_datepicker

**Comment by melloware** [comment]
I will double check what the other libs are doing but if you think about it from a screen reader perspective if the popup is open the screen reader is telling the user that. And they are interacting with the popup. If you are typing you are interacting with the Input. How will the screen reader know what to tell a visually impaired user?

**Comment by melloware** [comment]
Also see this note: https://github.com/primefaces/primefaces/pull/10513#issuecomment-1697508560

Material UI does the same thing we are doing. The difference is we have showOnFocus true by default and they have it false.

**Comment by melloware** [comment]
cc @Bukama @Rene-Stoeckel @christophs78 

OK did some research and I believe the current behavior is correct according to https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/

> Open the date picker dialog.  Move focus to selected date, i.e., the date displayed in the date input text field. If no date has been selected, places focus on the current date. 

here is the current 14.0 datepicker in the QF showcase: https://quarkus-faces-melloware-8a6a34c1.koyeb.app/ui/input/datepicker/datePickerJava8.xhtml?

I believe the proper example you want is Button

![image](https://github.com/primefaces/primefaces/assets/4399574/c1948123-5323-4748-b76a-f7bf1133e291)

But the legacy behavior PF had of typing while the Popup is open is not WCAG compliant.

**Comment by HSnicolascerrejon** [comment]
Hi! As I mentioned in #11951"it would be great if, after clicking the field when the popup is oppened, it would be closed and the field would still have focus" to let the manual date input. This could improve a lot the user experience while WCAG compliant.

Regards!

**Comment by melloware** [comment]
MonkeyPatch... now when you press ESC or click in the Input with your mouse when the panel is open it closes the panel and focuses the input allowing you

```js
if (PrimeFaces.widget.DatePicker) {
    $.prime.datePicker.prototype.onInputClick = function(event) {
        if (this.documentClickListener) {
            this.datepickerClick = true;
        }

        if (this.isPanelVisible()) {
            this.hideOverlay();
        } else if (this.options.showOnFocus) {
            this.showOverlay();
        }
    };
};
```

**Comment by Bukama** [comment]
> is this really a WCAG requirement? how do it other libs? very weird to disallow typing

Not really weird, but also not an exact WCAG req as @melloware has already pointed out. With the popup opens the focus has to go there and if you close it, it goes back to where you were (and not anywere on the page, which is a common mistake / programming error). So I like the "MonkeyPatch"

**Comment by melloware** [comment]
Thanks for making me look at this again. I actually found a few scenarios that were working while implementing this and was able to fix them all and improve the accessibility overall of the DatePicker!
